--- a/engines/e_devcrypto.c
+++ b/engines/e_devcrypto.c
@@ -887,7 +887,7 @@
     for (i = 0, known_digest_nids_amount = 0; i < OSSL_NELEM(digest_data);
         i++) {
 
-        selected_digests[i] = 1;
+        selected_digests[i] = 0;
 
         /*
          * Check that the digest is usable
@@ -1096,7 +1096,7 @@
 #ifdef IMPLEMENT_DIGEST
     { DEVCRYPTO_CMD_DIGESTS,
         "DIGESTS",
-        "either ALL, NONE, or a comma-separated list of digests to enable [default=ALL]",
+        "either ALL, NONE, or a comma-separated list of digests to enable [default=NONE]",
         ENGINE_CMD_FLAG_STRING },
 #endif
 
